### ISSUE TITLE
build exoplayer with FFMPEG extension

### ISSUE THREAD

**Issue by fgortiz** [TheWidlarzGroup/react-native-video#3106]
Is there a documented way to implement the ExoPlayer FFmpeg extension for this project?

**Comment by freeboub** [comment]
No, currently it is not possible. Exoplayer changed how they support it so now it is mandatory to build from exoplayer source. I think a patch in graddle is mandatory

**Comment by fgortiz** [comment]
In lieu of that, is there a list of audio codecs ExoPlayer (and therefore, RNV alpha) supports out of the box?

**Comment by freeboub** [comment]
Not so easy :)
The list of codec supported by exoplayer:
https://exoplayer.dev/supported-formats.html
But it can also depend of the android integration / version ...

**Comment by freeboub** [comment]
I created a PR to build the app from exoplayer source here: https://github.com/TheWidlarzGroup/react-native-video/pull/3932
It shall be a first step to build exoplayer with gstreamer !

**Comment by polaris018** [comment]
+1

**Comment by freeboub** [comment]
If someone need to implement it, here are the requiered steps

1/ build your app from expoplayer Sources. it is documented in react native video doc.
2/ you need to enable ffmpeg in the app: This is documented here: https://github.com/androidx/media/blob/release/libraries/decoder_ffmpeg/README.md
The first step is to allow building ffmpeg and to link it with exoplayer
I was able to reach these steps (I know it is possible to do it.)
3/ patch react native video to enable ffmpeg support.
This is the last section of the doc: " Using the module with ExoPlayer" the code change should be in function:

private MediaSource buildMediaSource(Uri uri, String overrideExtension, DrmSessionManager drmSessionManager, long cropStartMs, long cropEndMs) {

but, please validate it as a POC first I let see together the api changes to implement is needed !
Thank you

**Comment by bhadana-mayank** [comment]
> private MediaSource buildMediaSource(Uri uri, String overrideExtension, DrmSessionManager drmSessionManager, long cropStartMs, long cropEndMs) {

what you meant by this? do we need to do something here or below one?

DefaultRenderersFactory renderersFactory =
                new DefaultRenderersFactory(getContext())
                        .setExtensionRendererMode(DefaultRenderersFactory.EXTENSION_RENDERER_MODE_OFF)
                        .setEnableDecoderFallback(true)
                        .forceEnableMediaCodecAsynchronousQueueing();
